### PR TITLE
Add Monitoring v3 Samples

### PR DESCRIPTION
**Pull request by waprin**
Note: Need tests. I am working on a followup
commit to this and should have it ready by EOD.

You already reviewed this code, this is migration
from git-on-borg.

### CODE REVIEW

File: monitoring/list_resources.js [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+    if (error) {
+        console.log(error);
+        process.exit();
+    }
+
+    // Depending on the environment that provides the default credentials
+    // (e.g. Compute Engine, App Engine), the credentials retrieved may require
+    // you to specify the scopes you need explicitly.
+    // Check for this case, and inject the Cloud Storage scope if required.
+    if (authClient.createScopedRequired &&
```
Comment (jonparrott) [inline monitoring/list_resources.js]: Is this needed anymore? The Python client handles it and I thought I'd yelled about it in Node.js.

**Comment by jmdobry** [comment]
Lots of JSHint errors:

https://travis-ci.org/GoogleCloudPlatform/nodejs-docs-samples/jobs/116475506#L1420

File: monitoring/README.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+     [or click here ]{https://console.developers.google.com/project/_/apiui/credential/serviceaccount)
+      Select a JSON-format key, which will be downloaded to your computer. The key file must be located on the computer(s) from which you will access the Monitoring API. Access to the file should be restricted so that only trusted people can read the key.
+
+    ```
+    export GOOGLE_APPLICATION_CREDENTIALS=~/Downloads/<project-id>-0123456789abcdef.json
+    ```
+
+# Set up your local dev environment
+
+While Monitoring v3 is in EAP, the necessary client libraries are vendored in, so google-api-node-client is not
```
Comment (JustinBeckwith) [inline monitoring/README.md]: That's super weird.  We can publish the NPM module right now if that's the issue.

File: monitoring/list_resources.js (hunk 1 above)
Comment (jmdobry) [inline monitoring/list_resources.js]: I don't think it's done for you:

https://github.com/google/google-auth-library-nodejs#example-usage

File: monitoring/README.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+a package.json dependency. async was also vendored for convienence, so there is no need to run npm install unless
+you add new dependency.
+# Run locally
+
+    node list_resources.js <YOUR-PROJECT-ID>
+    node create_custom_metric.js <YOUR-PROJECT-ID>
+
+## Contributing changes
+
+See CONTRIBUTING.md
```
Comment (JustinBeckwith) [inline monitoring/README.md]: I don't think we need the contribution and licensing links here, it's covered at the top level repo

File: monitoring/create_custom_metric.js [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+            ],
+            metricKind: 'GAUGE',
+            valueType: 'INT64',
+            unit: 'items',
+            description: 'An arbitrary measurement.',
+            displayName: 'Custom Metric'
+        }
+    }, function (error, customMetric) {
+        if (error) {
+            console.log('Error Creating Custom Metric', error);
```
Comment (JustinBeckwith) [inline monitoring/create_custom_metric.js]: console.error

File: monitoring/README.md (hunk 2 above)
Comment (waprin) [inline monitoring/README.md]: ugg sorry yes README oversight, Jason already did it.

File: monitoring/list_resources.js (hunk 1 above)
Comment (waprin) [inline monitoring/list_resources.js]: Breaks when I remove it so still seems necessary.

File: monitoring/create_custom_metric.js (hunk 4 above)
Comment (waprin) [inline monitoring/create_custom_metric.js]: done in several places

File: monitoring/README.md (hunk 3 above)
Comment (waprin) [inline monitoring/README.md]: done

**Comment by waprin** [comment]
Ty for the extra review Justin/Jon (3 names that start with J in the review btw) I addressed all review comments, jshint passes. I think one of the Travis jobs is about to fail due to a flaky Datastore test. Lmk if you think I can merge anyway, trying to help techwriter get something to point to.

**Comment by JustinBeckwith** [comment]
Are we in a particular rush to have this in before there are tests?  If there's a good reason, that's cool, but otherwise I'd prefer to have tests with the PR.

**Comment by waprin** [comment]
Stackdriver is supposed to launch sometime soon. They are EST though and I haven't heard fromt hem for a while so maybe it will be tomorrow. I am working on the test code commit right now though should be done soon.

**Comment by waprin** [comment]
Ok, Travis is passing, but I think we're not in a rush after all. I'll finish the test commit then re-ping you guys, tahnks.

**Comment by waprin** [comment]
Maybe slight rush, tomorrow by 8:30 AM. I think we might just go live without Node though. Working on finishing up tests right now.

**Comment by jonparrott** [comment]
Merging for now due to urgency of release. @waprin gets an angry bug for tests.

**Comment by waprin** [comment]
Thanks. I'll try to get the testing PR up ASAP (hopefully tonight) just wanted to get ready for launch.

**Pull request merged**